src/server/scripts/EasternKingdoms/TheStockade/boss_hogger.cpp

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2008-2017 TrinityCore <http://www.trinitycore.org/>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "InstanceScript.h"
+#include "the_stockade.h"
+
+enum Says
+{
+    SAY_PULL       = 0, // Forest just setback!
+    SAY_ENRAGE     = 1, // AreaTriggerMessage: Hogger Enrages!
+    SAY_DEATH      = 2, // Yiipe!
+
+    SAY_WARDEN_1   = 0, // Yell - This ends here, Hogger!
+    SAY_WARDEN_2   = 1, // Say - He's...he's dead?
+    SAY_WARDEN_3   = 2 // Say - It's simply too good to be true. You couldn't have killed him so easily!
+};
+
+enum Spells
+{
+    SPELL_VICIOUS_SLICE     = 86604,
+    SPELL_MADDENING_CALL    = 86620,
+    SPELL_ENRAGE            = 86736
+};
+
+enum Events
+{
+    EVENT_VICIOUS_SLICE = 1,
+    EVENT_MADDENING_CALL,
+
+    EVENT_SAY_WARDEN_1,
+    EVENT_SAY_WARDEN_2,
+    EVENT_SAY_WARDEN_3
+};
+
+enum Points
+{
+    POINT_FINISH = 0
+};
+
+Position const wardenthelwatermovepos = { 152.019f, 106.198f, -35.1896f, 1.082104f };
+Position const wardenthelwaterpos = { 138.369f, 78.2932f, -33.85627f, 1.082104f };
+
+class boss_hogger : public CreatureScript
+{
+public:
+    boss_hogger() : CreatureScript("boss_hogger") {}
+
+    struct boss_hoggerAI : public BossAI
+    {
+        boss_hoggerAI(Creature* creature) : BossAI(creature, DATA_HOGGER) 
+        {
+            hasEnraged = false;
```

Comment:
Just codestyle, you dont need change if you dont want
but it can be changed by: ```cpp boss_hoggerAI(Creature* creature) : BossAI(creature, DATA_HOGGER), hasEnraged(false) { } ```